scrapy_poet/downloadermiddlewares.py

```diff
@@ -49,14 +50,25 @@ def __init__(self, crawler: Crawler) -> None:
         """Initialize the middleware"""
         self.crawler = crawler
         settings = self.crawler.settings
-        registry_cls = load_object(
-            settings.get("SCRAPY_POET_OVERRIDES_REGISTRY", OverridesRegistry)
-        )
-        self.overrides_registry = create_instance(registry_cls, settings, crawler)
+
+        if settings.get("SCRAPY_POET_OVERRIDES_REGISTRY"):
+            msg = (
+                "The SCRAPY_POET_OVERRIDES_REGISTRY setting is deprecated. "
+                "Use SCRAPY_POET_REGISTRY instead."
+            )
+            warnings.warn(msg, DeprecationWarning, stacklevel=2)
+            registry_cls = load_object(
+                settings.get("SCRAPY_POET_OVERRIDES_REGISTRY", OverridesAndItemRegistry)
```

Comment:
At this point we know it is defined.

```suggestion
                settings["SCRAPY_POET_OVERRIDES_REGISTRY"]
```